Epsilon/Service/FilterService.cs

```diff
@@ -0,0 +1,56 @@
+using DocumentFormat.OpenXml;
+using DocumentFormat.OpenXml.Packaging;
+using DocumentFormat.OpenXml.Wordprocessing;
+using Epsilon.Abstractions.Component;
+using Epsilon.Abstractions.Service;
+using Epsilon.Canvas.Abstractions.Model;
+using Epsilon.Canvas.Abstractions.Model.GraphQl;
+using Epsilon.Canvas.Abstractions.Service;
+using Microsoft.Extensions.Configuration;
+
+namespace Epsilon.Service;
+
+public class FilterService : IFilterService
+{
+    private const string GetUserSubmissions = @"
+        query GetUserSubmissions {
+          allCourses {
+            submissionsConnection(studentIds: $studentIds) {
+              nodes {
+                submittedAt
+              }
+            }
+          }
+        }
+    ";
+
+    private readonly IConfiguration _configuration;
+    private readonly IAccountHttpService _accountHttpService;
+    private readonly IGraphQlHttpService _graphQlService;
+
+    public FilterService(IConfiguration configuration, IAccountHttpService accountHttpService, IGraphQlHttpService graphQlHttpService)
+    {
+        _configuration = configuration;
+        _accountHttpService = accountHttpService;
+        _graphQlService = graphQlHttpService;
+    }
+
+    public async Task<IEnumerable<EnrollmentTerm>> GetParticipatedTerms()
+    {
+        var allTerms = await _accountHttpService.GetAllTerms(1);
+
+        var studentId = _configuration["Canvas:StudentId"];
+        var submissionsQuery = GetUserSubmissions.Replace("$studentIds", $"{studentId}", StringComparison.InvariantCultureIgnoreCase);
+
+        var response = await _graphQlService.Query<CanvasGraphQlQueryResponse>(submissionsQuery);
+        var submissions = response!.Data!.Courses!.SelectMany(c => c.SubmissionsConnection!.Nodes);
```

Comment:
Instead of skipping nullability, we should do some proper null checks here on the  response. The ones in the query you can ignore, but the response is important to check,